packages/maps/src/Client.ts

```diff
@@ -53,7 +53,62 @@ export class Maps {
     return this.instantiateMap(mapConfig);
   }
 
-  private async instantiateMap(mapConfig: unknown) {
+  public static generateMapConfigFromSource(source: string) {
+    const type = source.search(' ') > -1 ? 'sql' : 'dataset';
+
+    return {
+      [type]: source,
+      vectorExtent: 2048,
```

Comment:
We should probably extract 2048 from this method and also equivalent ones on `instantiateMapFrom` into some shared consts